app/models/spree/shipping_method_decorator.rb

```diff
@@ -49,14 +49,6 @@ def available_to_order_with_distributor_check?(order, display_on=nil)
   end
   alias_method_chain :available_to_order?, :distributor_check
 
-  def within_zone?(order)
-    if order.ship_address
-      zone && zone.include?(order.ship_address)
-    else
-      true # Shipping methods are available before we've selected an address
-    end
-  end
```

Comment:
I found out that this method is used internally by spree in `core/app/models/spree/shipping_method.rb`

```ruby
def available_to_order?(order, display_on= nil)
  available?(order, display_on) &&                                    
  within_zone?(order) &&           
  category_match?(order) &&        
  currency_match?(order)                                                                                                                    
end    
```

Although by that comment in line 56, I think we overrode that method so that we actually show the shipping method before the user specifying a shipping adress as by doing so `available_to_order?` will return true.

Turns out that we only introduced that `else` branch as the spree's implementation is exactly the same:

```ruby
def within_zone?(order)    
  zone && zone.include?(order.ship_address)                   
end
```

To me it seems that we implemented the behaviour in the wrong place, changing the meaning of `within_zone?`. IMO it should be done in the view layer since this has nothing to do about business logic but UI. Having less overrides is always better.

Reply:
Talked offline with @sauloperez and @oeoeaio , we agreed that we don't need `#within_zone` customization anymore.